### ISSUE TITLE
bidirectional_gru_bn_layer问题

### ISSUE THREAD

**Issue by bliunlpr** [PaddlePaddle/DeepSpeech#86]
在 model_utils/network.py 107行 , 定义了 bidirectional_gru_bn_layer(name, input, size, act) 这个函数，做的是双向带bn的gru。在程序里，先做了一个线性变换层，应该是 输入乘以一个权重矩阵（代表了gru计算公式中的前半部分，也就是 W_z * X, W_r * X, W * X）。紧接着做了bn。但是在计算bn的时候，会先做一个 减均值除标准差 的操作，如果输入乘了权重矩阵，再进行 减均值除标准差的操作 的时候，结果是一样的。 换句话说，上面的线性变换层是不是就失去作用了？无论变换是什么，计算bn 的时候都抵消了。不知道我这样理解是不是有问题？诚心求教，谢谢！

**Comment by pkuyym** [comment]
batch norm一般会加在线性变换后，非线性激活前，这里的用法符合这个习惯（非线性计算在gru里面）
线性变换跟减均值除方差（沿channel方向）应该是有gap的
RNN里面用BN，可以看论文的描述

> Recent research has shown that BatchNorm improves the speed of convergence of recurrent nets, without showing any improvement in generalization performance [37]. In contrast, we demonstrate that when applied to very deep networks of simple RNNs on large data sets, batch normalization substantially improves final generalization error while greatly accelerating training.
